controllers/snapshot/snapshot_adapter.go

```diff
@@ -249,7 +271,15 @@ func (a *Adapter) EnsureStaticIntegrationPipelineRunsExist() (controller.Operati
 
 		err = gitops.WriteIntegrationTestStatusesIntoSnapshot(a.snapshot, testStatuses, a.client, a.context)
 		if err != nil {
-			return controller.RequeueWithError(err)
+			a.logger.Error(err, "Failed to update test status in snapshot annotation")
+			errsForPLRCreation = errors.Join(errsForPLRCreation, err)
+		}
+
+		if errsForPLRCreation != nil {
+			if clienterrors.IsInvalid(errsForPLRCreation) {
```

Comment:
I'd say, we want to stop only if all errors are invalid. Because they might we a test we failed to run due to an intermittent error

Reply:
Here `controller.StopProcessing()` is called when where is invalid resource error in the errors, otherwise `controller.RequeueWithError(errsForPLRCreation)` is called below. Because as long as there is an invalid resource error, manual action is needed.
Then let me exchange their order.

Reply:
Updated. Thanks.

Reply:
I don't know what this fixes, it's IMO just invariant
![image](https://github.com/redhat-appstudio/integration-service/assets/20207783/539e2d5b-40fd-4bd2-bd65-d6fdf9ab99ca)


What I meant, that if we cannot create a test due an intermittent issue, we won't retry.

So basically, it's still that one invalid test may block creation of others tests because we won't retry on intermittent issue.

Reply:
Do you mean this PR should figure out the intermittent issue and retry for them? If yes, I think so far we can figure out all intermittent issue. The current code will retry if non-invalid error is not included in it but stop if the invalid error is included.
We may need to pay some attention to the integration log to see if there is any more intermittent issue or unrecoverable error, then add code for them.
Do you have any good idea to figure them out now?

Reply:
We have N tests, if 1 tests fail on invalid error, why we should stop reconciliation? There can be another failed test that could just failed to be created on intermittent error.

Also, if test is invalid, and it's only optional tests, do we want to stop reconciliation in this case?

I don't like handling of tests in general here, we should try to create all tests, record their status/failures and continue with reconciliation, user can restart tests if needed.
This retry/stop reconciliation logic seems broken to me and just add more work and uncertainty.

Also this story is spike, it was supposed to be investigation how we can handle test parallelization better, why am I reviewing PR?

Reply:
> We have N tests, if 1 tests fail on invalid error, why we should stop reconciliation? There can be another failed test that could just failed to be created on intermittent error.
> 
In the updated code, if the errors doesn't include invalid error, we will retry with error.
> Also, if test is invalid, and it's only optional tests, do we want to stop reconciliation in this case?
> 
We never mentioned the optional tests in this ticket's AC. I think it is another story/spike including more code change, even about how we should handle the failed optional integration test in pr and push event.
> I don't like handling of tests in general here, we should try to create all tests, record their status/failures and continue with reconciliation, user can restart tests if needed. This retry/stop reconciliation logic seems broken to me and just add more work and uncertainty.
> 
It is trying to create all tests, record the status/failure in snapshot annotation, users can restart the test by rerun label or trigger on UI.
> Also this story is spike, it was supposed to be investigation how we can handle test parallelization better, why am I reviewing PR?
From my point of view, even for a spike, we can still gather review/comment/idea/feedback to find a good solution.

Reply:
> > We have N tests, if 1 tests fail on invalid error, why we should stop reconciliation? There can be another failed test that could just failed to be created on intermittent error.
> 
> In the updated code, if the errors doesn't include invalid error, we will retry with error.

That's what I mean, still if there is one test that's invalid, it prevents other tests from being retried

Reply:
> > > We have N tests, if 1 tests fail on invalid error, why we should stop reconciliation? There can be another failed test that could just failed to be created on intermittent error.
> > 
> > 
> > In the updated code, if the errors doesn't include invalid error, we will retry with error.
> 
> That's what I mean, still if there is one test that's invalid, it prevents other tests from being retried

Ah, yes, `!clienterrors.IsInvalid(errsForPLRCreation)` should be updated to check none of the error is invalid error. Let me think. Thanks.